cmd/orch-tester/orch_tester.go

```diff
@@ -549,3 +599,209 @@ func (s *streamerClient) getOrchestrators() ([]*orch, error) {
 
 	return data.Data.Transcoders, nil
 }
+
+// Get a random source/rendition sample segment and write it to external storage
+func (s *streamerClient) randomSample(ctx context.Context, mid string, orch string) error {
+	randSeqNo := rand.Int() % numSegments
+	randProfile := s.profiles[rand.Int()%len(s.profiles)]
+	//stream/manifestID/profile/SeqNo.ts
+	sourceUrl := fmt.Sprintf("%v/stream/%v_0_0/source/%v.ts", s.broadcaster, mid, randSeqNo)
+	renditionUrl := fmt.Sprintf("%v/stream/%v_0_0/%v/%v.ts", s.broadcaster, mid, randProfile, randSeqNo)
+	var (
+		rendition []byte
+		source    []byte
+		err       error
+	)
+	tick := time.NewTicker(2 * time.Second)
+	for {
+		select {
+		case <-ctx.Done():
+			glog.Info("random sampler timed out")
+			// If we haven't found the segment try to sample from whatever playlist we can
+			rendition, source, renditionUrl, err = s.sampleFromPlaylist(mid)
+			if err != nil {
+				glog.Error(err)
+			}
+			break
+		case <-tick.C:
+			glog.V(model.DEBUG).Infof("Trying to download %v", renditionUrl)
+			rendition, err = s.downloadSegment(renditionUrl)
+			if err != nil {
+				glog.V(model.DEBUG).Info(err)
+				continue
+			}
+			source, err = s.downloadSegment(sourceUrl)
+			if err != nil {
+				glog.V(model.DEBUG).Info(err)
+				continue
+			}
+			break
+		}
+		break
+	}
+
+	if len(rendition) == 0 || len(source) == 0 {
+		return fmt.Errorf("no segments found")
+	}
+
+	y, m, d := start.Date()
+	baseFileName := fmt.Sprintf("%v-%v%v", y, m.String(), d)
+	u := strings.Split(renditionUrl, "/")
+	u = u[len(u)-2:]
+	rendF := fmt.Sprintf("%v/%v/%v-%v", baseFileName, orch, mid, strings.Join(u, "-"))
```

Comment:
Format -> `date/orch/manifestID/rendition/seqNo.ts` to follow a similar file path convention as the B?

Reply:
I felt like this would make the directory structure deeper than necessary , it would create two more folders for both the source and the rendition to only contain those files respectively.